app/models/compound_aliquot.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class CompoundAliquot # rubocop:todo Style/Documentation
+  include ActiveModel::Model
+
+  DUPLICATE_TAG_DEPTH_ERROR_MSG = "Cannot create compound sample from following samples due to duplicate 'tag depth'"
+  MULTIPLE_STUDIES_ERROR_MSG =
+    'Cannot create compound sample due to the component samples being under different studies.'
+
+  attr_accessor :request, :source_aliquots
+
+  attr_reader :component_samples
+
+  validate :tag_depth_is_unique
+
+  def initialize(attributes)
+    super
+
+    @component_samples = source_aliquots.map(&:sample)
+  end
+
+  # Check that the component samples in the compound sample will be able to be distinguished -
+  # this is represented by them all having a unique 'tag_depth'
+  def tag_depth_is_unique
+    return unless source_aliquots.pluck(:tag_depth).uniq.count != source_aliquots.size
+
+    errors.add("#{DUPLICATE_TAG_DEPTH_ERROR_MSG}: #{component_samples.map(&:name)}")
+  end
+
+  def default_library_type
+    source_aliquots.first.library_type
+  end
+
+  # Default study that the new compound sample will use
+  # Uses the one from the request if it's present,
+  # otherwise, the one from the source aliquots if it's consistent.
+  def default_compound_study
+    request.initial_study ||
+      begin
+        raise MULTIPLE_STUDIES_ERROR_MSG if studies.count > 1
+
+        studies.first
+      end
+  end
+
+  def studies
+    source_aliquots.map(&:study).uniq
+  end
+
+  # Default project that the new compound sample will use
+  # Uses the one from the request if it's present,
+  # otherwise, one grabbed from a source aliquot.
+  def default_compound_project_id
+    request.initial_project_id || source_aliquots.first.project_id
```

Comment:
Perhaps we want the same check for studies as for projects.

Reply:
I don't know. I don't know much about projects and where they're used, other than the fact that they're used in billing. Will it matter which project id ends up on the compound sample aliquot?

Reply:
Oh dear... we might need to check this.

So usually projects are billed based on the fraction of a lane their aliquots take up, with the assumption that all aliquots are loaded at approximately equal concentrations. So if a single project was assigned to a compound aliquot, they'd be billed, but the other projects would get it for free.

But the additional gotcha, is that I'm not sure its true to assume that a compound aliquot it the same 'size' as other aliquots. Billing reports will need to re-inflate them.